Week8/app/src/main/java/com/bartoszkrol/catfacts/worker/DownloadDataWorker.kt

```diff
@@ -0,0 +1,35 @@
+package com.bartoszkrol.catfacts.worker
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.bartoszkrol.catfacts.App
+import com.bartoszkrol.catfacts.database.CatFactsDatabase
+import com.bartoszkrol.catfacts.database.RoomRepository
+import com.bartoszkrol.catfacts.model.Success
+
+class DownloadDataWorker(context: Context, params: WorkerParameters) : CoroutineWorker(context, params) {
+
+    private val remoteApi = App.remoteApi
+
+    companion object {
+        const val DOWNLOAD_DATA_WORKER = "DOWNLOAD_DATA_WORKER"
+    }
+
+    override suspend fun doWork(): Result {
+        val result = remoteApi.getCatFacts()
+
+        if (result is Success) {
+            // Add new data to the database
+            // LiveData will update recyclerView with a new facts about the cats
+            val catFactsDao = CatFactsDatabase.getDatabase(applicationContext).catFactsDao()
```

Comment:
It would be better if you had factory functions for these things, so you don't have to do this manually here!

Reply:
Now dao and repository are available from the app class.